app/models/user.rb

```diff
@@ -1,4 +1,18 @@
 class User < ApplicationRecord
   attr_accessor :display_name, :admin, :roles
   validates :login, presence: true
+
+  def self.from_jwt(data = { })
+    id, login = data&.values_at 'id', 'login'
+    return unless id && login
+
+    User.where(id: id, login: login).first_or_create do |user|
+      user.display_name = data['dname']
+      user.admin = data['admin'] == true # explicit
+    end
+  end
+
+  def roles
+    @roles || { }
```

Comment:
What if the user had their admin access revoked? Implementing storage+caching/expiration of tokens is something I can open an issue for and we can discuss there.

Reply:
sure if they have revoked roles caching is an issue. I think you'll see varying levels of performance when trying to enumerate these roles, especially for users with lots of roles and specifically when accessing these roles without the project context.

This can be dealt with outside the scope of this PR, please do file and issue for it and test for it.

Reply:
https://github.com/zooniverse/tove/issues/27